src/templates/sermons.factory.js

```diff
@@ -25,16 +31,42 @@ exports.createPages = async (graphql, createPage) => {
     }
 }
 `)
-        const nodes = result &&
-            result.data &&
-            result.data.avorg &&
-            result.data.avorg.sermons &&
-            result.data.avorg.sermons.nodes
-
-        await createPage({
-            path: `${lang.base_url}/sermons`,
-            component: path.resolve(`./src/templates/sermons.js`),
-            context: {nodes}
-        })
+
+    const sermons = _.get(result, 'data.avorg.sermons'),
+        nodes = _.get(sermons, 'nodes'),
+        sermonCount = _.get(sermons, 'aggregate.count', 0)
+
+    await createPage({
+        path: `${lang.base_url}/sermons/page/${page}`,
+        component: path.resolve(`./src/templates/sermons.js`),
+        context: {
+            nodes,
+            pagination: {
+                total: Math.ceil(sermonCount / 10),
+                current: page
+            }
+        }
+    })
+
+    const hasNextPage = _.get(sermons, 'pageInfo.hasNextPage')
+
+    if (hasNextPage) {
+        const endCursor = _.get(sermons, 'pageInfo.endCursor')
+
+        await createPagesByLang(
+            langKey,
+            graphql,
+            createPage,
+            page + 1,
+            endCursor
+        )
+    }
+}
+
+exports.createPages = async (graphql, createPage) => {
+    const langKeys = Object.keys(constants.languages)
+
+    await Promise.all(langKeys.map(async (key) => {
+        return await createPagesByLang(key, graphql, createPage)
```

Comment:
You either want `return` or `await` here, not both. I would do it like this:

```js
await Promise.all(langKeys.map((key) => createPagesByLang(key, graphql, createPage)));
```

Reply:
Haha, I wondered why that change had absolutely no effect on build times